[MIG] sale_automatic_workflow: Migration to 14.0

phuctranfxvn opened the pull request.

sebastienbeau committed:
[ADD] add sale_automatic_workflow, a module to automatized the sale order workflow

sebastienbeau committed:
[IMP] sale_automatic_workflow : add default value for workflow method

Commit:
[FIX] sale_automatic_workflow : fix invoice creation from picking

Commit:
[IMP] sale_automatic_workflow: redesign of sale order view

sebastienbeau committed:
[FIX] sale_automatic_workflow : add missing security rule

sebastienbeau committed:
[FIX] sale_automatic_workflow : reconciled should be done by the cron task, indeed if we did it in the workflow we can not anymore unreconcile an invoice because it will be re-reconcile directly

sebastienbeau committed:
[FIX] sale_automatic_wokflow : fix the option force invoice date, indeed this feature was still not reimplemented

sebastienbeau committed:
[REF] refactor/clean code due to orm change in 6.1

sebastienbeau committed:
[IMP] sale_automatic_workflow : use context manager for intermediary commit

sebastienbeau committed:
[IMP] sale_automatic_workflow : add doc string for commit_now and set to False params raise_error as it's a boolean

Commit:
Ir.cron protected by a noupdate flag.

sebastienbeau committed:
[IMP] sale_automatic_workflow : invoice validation and reconciliation should be process individualy

sebastienbeau committed:
[REF] import the commit_now from framework_helpers as it have been extracted

sebastienbeau committed:
[FIX] Open a new cursor before commiting it, thanks to Guewen ideas

sebastienbeau committed:
[IMP] set correctly the version number 6.1.X

sebastienbeau committed:
[IMP] sale_automatic_workflow: improve invoice reconciliation by supporting exchange write off

sebastienbeau committed:
[MIGR] sale_automatic_workflow: move code in the correct file

sebastienbeau committed:
[MIGR] sale_automatic_workflow: migration module to V7

Commit:
[FIX] sale_automatic_workflow: replace dependencies on sale_quick_payment by sale_quick_payment

Commit:
[IMP] sale_automatic_worflow: reconcile with the move lines instead of the voucher

Commit:
[IMP] sale_automatic_workflow: review

Commit:
[FIX] since openerp 7, the crons callee have their own cr, so we don't need anymore to open a new one

Commit:
[IMP] review the workflow job, get rid of the framework helpers dependency

Commit:
[FIX] remove the field 'validate_manufactoring_order', not implemented actually

Commit:
[IMP] use an onchange to apply the default values (connectors will have to use this onchange

Commit:
[FIX] rename onchange_workflow_process to onchange_workflow_process_id

Commit:
[IMP] sale_automatic_workflow: use the same words than openerp for order_policy, remove the 'postpaid' removed in openerp

Commit:
[IMP] sale_automatic_workflow: display the workflow below the payment method

Commit:
[IMP] sale_automatic_workflow: the onchange apply the workflow

Commit:
[FIX] forbid the unlink of a workflow used by a sale order

Commit:
[REF] sale_automatic_workflow: remove the field 'days_before_cancel', moved in connector_ecommerce

Commit:
[CHG] sale_automatic_workflow: 'validate_order' is now a boolean, because the orders are now created or not based on the payment status

Commit:
[FIX] noupdate on the automatic workflow

Commit:
[FIX] the workflow_process_id is not transferred to picking, due to a big and incomprehensible jiggery-pokery between stock.picking and stock.picking.out.

The field have to be created on stock.picking so the sale order is able to write the field on creation (in sale._create_pickings_and_procurements), but thereby we can't search a stock.picking.out by this field.

Commit:
[ADD] sale_automatic_workflow: automatic validation of sales orders according to the workflow

Commit:
[IMP] better field descriptions (respect the sale order ones) and change the view inheritance due to changes in sale_payment_method

Commit:
[FIX] _prepare_invoice is called on stock.picking and not on stock.picking.out

Commit:
[IMP] rename Sale Workflow Method to Automatic Workflow

Commit:
[IMP] the option to create an invoice now allows to choose if it is created on the confirmation of the sale order or after the delivery

Commit:
[FIX] updated base data according to last change and addded a manual workflow

Commit:
[FIX] do not activate force invoice date in the base workflows

Commit:
[ADD] sale_automatic_workflow: french translation

sebastienbeau committed:
[FIX] when a sale order is in exception we should stop to try to validate it every minutes, performance killer. So the best is to only try to validate the sale order that don't have any exception

Commit:
[FIX] use explicit relative import, use standard orm.Model

Commit:
[FIX] the exception variable (e) is not used

Commit:
[ADD] authorship

Commit:
[FIX] unused import

Commit:
[CHG] use isinstance rather than hasattr

Commit:
[IMP] docstring of sale.workflow.process

Commit:
[FIX] fr_FR.po files not loaded, should be named fr.po

sebastienbeau committed:
[REF] adapte sale_automatic_workflow due to the change as sale order are linked to a account move line

Commit:
[ADD] Camptocamp as author alongside Akretion

sebastienbeau committed:
[FIX] remove dependency on sale_exception and give the posibility to inherit easily the domain for the sale order validation

Commit:
[FIX] missing argument in _get_sum_... methods, sale_order.payment_ids is already move lines, so we should not get their 'line_id'

Commit:
use method to get the move lines

Tranlations es for modules

Commit:
set the flag 'installable' to False on the modules

Commit:
move modules to migrate in __unported__

Commit:
Move sale_automatic_workflow from __unported__ to root folder

Commit:
Set sale_automatic_workflow to installable

Commit:
Extract module's description in README.rst

Commit:
Remove dependency on sale_payment_method

I want to remove this dependency so we can use an automatic workflow
without using the payment methods.

A third module will surely be necessary.

Commit:
Remove changes to the model payment.method

Commit:
Remove what depends on sale_payment_method

It will be reimplemented later in a link module between
sale_payment_method and sale_automatic_workflow

Commit:
Started to implement a test

Commit:
stock.picking.out has passed away and methods have changed

Commit:
Add a page for the Automation information

Commit:
Deprecated comment

Commit:
Test fails on the last assert

Because workflow_process_id is not propagated to the stock.picking

Commit:
Check if invoice is generated and opened

Commit:
Propagate the workflow to the picking

Test pass now

Commit:
New API for sale.workflow.process

Commit:
New api account.invoice

Commit:
New API for sale.order

Commit:
New API for stock.picking

Commit:
Rename stock.py to stock_picking.py

Commit:
remove dead xml

Commit:
New api for automatic.workflow.job

Commit:
Add tests for propagation of date and workflow

Commit:
Put fields in groups so the labels are displayed

Commit:
Updated pot and fr.po

Commit:
Unused import

Commit:
Typo, added an uppercase

Commit:
Do not propagate sale sale automatic workflow when several orders are grouped in one picking

If we have several sales orders in one procurement group,
we group them in one picking. The possibilities when we have
several orders for one picking are:
1. take the workflow of the first sale order
2. do not propagate the workflow at all
3. propagate the workflow if all the workflows are the same
The solution 1. is bad, the 3. could be hard to understand and
unpredictable. The solution 2. takes no assumption on what is
expected.

gurneyalex committed:
Add OCA as author of OCA addons

In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.

bguillot committed:
[FIX] change called method to confirm sale orders

yvaucher committed:
Add missing default oca icons

damdam-s committed:
[IMP] add relation to sales team

damdam-s committed:
[IMP] add group base.group_multi_salesteams on 'section_id' fields in views

sbidoul committed:
[UPD] prefix versions with 8.0

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

Commit:
sale_automatic_workflow : add default journal on workflow process

Commit:
sale_automatic_workflow : create invoice for sale with only product service

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

xavier-dass committed:
[WIP] : Automated Workflows

atchuthan committed:
[FIX] fixed conventions

atchuthan committed:
[FIX] modified data file

atchuthan committed:
[FIX] corrections and modified test

atchuthan committed:
[FIX] modified filter usage

atchuthan committed:
[FIX] modified filter usage

atchuthan committed:
[FIX] separated payment portion

atchuthan committed:
[FIX] corrections

atchuthan committed:
[MVD] moved payment filter to sale_automatic_workflow_payment_mode

Commit:
Fix default filters

The addon does not install if the table sale_workflow_process contains
records without filters, Odoo tries to get the xmlids of the default
values before the filters are created.

Commit:
Disable 'Transfer Picking' on automatic workflow

Back in 7.0, this option was not active by default on the automatic
workflow and I think it was better: this option is useful for some use
cases but is not really the *default* thing we want to do on an
automatic workflow.

oca-transbot committed:
OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

Commit:
[FIX] RST formatting

sbidoul committed:
[FIX] remove en.po that was erroneously created by transbot

oca-transbot committed:
OCA Transbot updated translations from Transifex

florian-dacosta committed:
Add possibility to invoice service automatically on delivery

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Rename manifest files

oca-transbot committed:
OCA Transbot updated translations from Transifex (+5 squashed commits)
Squashed commits:
[1b33e05] OCA Transbot updated translations from Transifex
[997016e] OCA Transbot updated translations from Transifex
[6aebc34] OCA Transbot updated translations from Transifex
[cc80052] OCA Transbot updated translations from Transifex
[87076c7] OCA Transbot updated translations from Transifex

Commit:
Option to set sale order done consider shipping

The option to automatically set the sales order as 'done' currently
filters only the sales orders in state 'sale' and invoiced.

I propose to change the default filter to 'all is invoiced AND all is
delivered' which emulates the behaviour of the previous versions of
Odoo.

This is a backward compatible change: as the filter has `noupdate="1"`,
existing installations will keep their domains and new installations
will get the new one.

pedrobaeza committed:
sale_*: Fix parent menu

The previous parent menu has an action attached, and with these modules,
it can't be clicked.

oca-transbot committed:
OCA Transbot updated translations from Transifex (+1 squashed commit)
Squashed commits:
[04700d6] OCA Transbot updated translations from Transifex

lasley committed:
[MIG] sale_automatic_workflow: Upgrade to v10
* Bump version
* Change openerp references to odoo
* Fix license headers
* Move menu
* Alphabetize imports
* Remove data key from XML
* Change deprecated invoice validation workflow call to replacement action method

lasley committed:
Fix API type & method name for picking values

lasley committed:
Use related fields for domains & char_domain where applicable

lasley committed:
Make domain fields invisible when no filter selected

lasley committed:
Switch commit to save point

lasley committed:
Fix lint

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[IMP] sale_automatic_workflow_payment_mode: improve test coverage

lmignon committed:
[IMP] sale_automatic_workflow_payment_mode: OCA conventions

oca-transbot committed:
OCA Transbot updated translations from Transifex

grindtildeath committed:
sale_automatic_workflow support multicompany

oca-transbot committed:
OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

grindtildeath committed:
Use force_company as contextmanager

grindtildeath committed:
Add multicompany unittest

grindtildeath committed:
Extend multicompany test with daughter company

grindtildeath committed:
Improve test with another company and account.journal assertion

grindtildeath committed:
Fix test (every company should be allowed)

jaredkipe committed:
[FIX] sale_automatic_workflow: multi-record aware on validation (#570)

* FIX automatic validation of pickings if the domain returns more than one

* Change picking validation to be transactionally similar to other jobs.

oca-transbot committed:
OCA Transbot updated translations from Transifex

gdgellatly committed:
[PORT] Sale Automatic Workflow to v11.0

oca-transbot committed:
OCA Transbot updated translations from Transifex

hugosantosred committed:
[FIX] Fixed tests for sale_automatic_workflow

With this commit in odoo-server https://github.com/OCA/OCB/commit/7ed86b53cdfbab9a7867de328604850138692caf#diff-100ea30dc07c0edef9040b4ef74279be
the moves that where force-assigned are not transferred in the picking, so in order to fix test is necessary to add stock to the stockable products in test sale order

oca-travis committed:
[UPD] Update sale_automatic_workflow.pot

Commit:
[12.0][MIG] sale_automatic_workflow: Migrate to 12.0

gdgellatly committed:
Fix some of the porting of sale_automatic_workflow [WIP]

sebalix committed:
[MIG] sale_automatic_workflow: Migration to 12.0

oca-travis committed:
[UPD] Update sale_automatic_workflow.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_automatic_workflow
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_automatic_workflow/

OCA-git-bot committed:
[UPD] README.rst

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (55 of 55 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_automatic_workflow
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_automatic_workflow/zh_CN/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (55 of 55 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_automatic_workflow
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_automatic_workflow/zh_CN/

tfossoul committed:
[IMP] : black, isort

tfossoul committed:
[MIG] sale_automatic_workflow: Migration to 13.0

oca-travis committed:
[UPD] Update sale_automatic_workflow.pot

OCA-git-bot committed:
[UPD] README.rst

rousseldenis committed:
[13.0][FIX] sale_automatic_workflow: Changed method to create invoices

Due to : https://github.com/odoo/odoo/commit/f808f36f3843875f0b87ab191277970564201429

simahawk committed:
sale_automatic_workflow: improve tests

OCA-git-bot committed:
sale_automatic_workflow 13.0.1.0.1

OCA-git-bot committed:
sale_automatic_workflow 13.0.1.0.2

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-13.0/sale-workflow-13.0-sale_automatic_workflow
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-13-0/sale-workflow-13-0-sale_automatic_workflow/

OCA-git-bot committed:
pre-commit update

dong-z committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 96.4% (53 of 55 strings)

Translation: sale-workflow-13.0/sale-workflow-13.0-sale_automatic_workflow
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-13-0/sale-workflow-13-0-sale_automatic_workflow/zh_CN/

MiquelRForgeFlow committed:
*: E741 ambiguous variable name 'l'

OCA-git-bot committed:
sale_automatic_workflow 13.0.1.0.3

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 98.2% (54 of 55 strings)

Translation: sale-workflow-13.0/sale-workflow-13.0-sale_automatic_workflow
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-13-0/sale-workflow-13-0-sale_automatic_workflow/zh_CN/

danimv5 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (55 of 55 strings)

Translation: sale-workflow-13.0/sale-workflow-13.0-sale_automatic_workflow
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-13-0/sale-workflow-13-0-sale_automatic_workflow/es/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-13.0/sale-workflow-13.0-sale_automatic_workflow
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-13-0/sale-workflow-13-0-sale_automatic_workflow/

kittiu committed:
[13.0][IMP] sale_automatic_workflow, add _do function

brendapaniagua committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (55 of 55 strings)

Translation: sale-workflow-13.0/sale-workflow-13.0-sale_automatic_workflow
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-13-0/sale-workflow-13-0-sale_automatic_workflow/ca/

OCA-git-bot committed:
sale_automatic_workflow 13.0.1.1.0

ThomasBinsfeld committed:
[FIX] sale_automatic_workflow: wrong model in _do_sale_done

britoederr committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (55 of 55 strings)

Translation: sale-workflow-13.0/sale-workflow-13.0-sale_automatic_workflow
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-13-0/sale-workflow-13-0-sale_automatic_workflow/pt_BR/

britoederr committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (55 of 55 strings)

Translation: sale-workflow-13.0/sale-workflow-13.0-sale_automatic_workflow
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-13-0/sale-workflow-13-0-sale_automatic_workflow/pt_BR/

OCA-git-bot committed:
sale_automatic_workflow 13.0.1.1.1

phuctranfxvn committed:
[IMP] sale_automatic_workflow: black, isort, prettier

atchuthan commented:
Please check the old PR https://github.com/OCA/sale-workflow/pull/1321, Please review it, and let's merge

nilshamerlinck commented:
> Please check the old PR #1321, Please review it, and let's merge

Hi @atchuthan oups, seems this previous PR was missed, probably because it was not announced in https://github.com/OCA/sale-workflow/issues/1274 :/

To be pragmatic, given that https://github.com/OCA/sale-workflow/pull/1321 is red and has some fixes pending while this one passes and works, I think this one should be merged

florian-dacosta commented on sale_automatic_workflow/models/stock_picking.py:
```diff
@@ -0,0 +1,36 @@
+                ):
+                    for move_line in move.move_line_ids:
+                        move_line.qty_done = move_line.product_uom_qty
+            picking.with_context(skip_overprocessed_check=True).button_validate()
```
Context `skip_overprocessed_check` does not exist anymore in stock module. I think `skip_immediate` should be used instead

phuctranfxvn replied:
Yes, you are right, skip_overprocessed_check is deprecated. fixed in the latest update

phuctranfxvn committed:
[MIG] sale_automatic_workflow: Migration to 14.0

atchuthan reviewed (approved):
Code review :+1:

atchuthan commented:
@florian-dacosta Please check the changes and update the review status

grindtildeath committed:
sale_automatic_workflow: Do not post invoices set back to draft

When sale automatic workflow is set to create and post invoices,
a posted invoice that is set back to draft by the user will be
posted again automatically on the next run of the sale automatic
workflow.

This change in the filter avoids to post the invoice automatically
and allows the user to do its changes before posting it manually.

grindtildeath commented:
@phuctranerp Hello, can you please cherry-pick https://github.com/OCA/sale-workflow/pull/1492 ?
I guess the domain needs to be changed for v14.0 by using new field `posted_before`: https://github.com/odoo/odoo/blob/14.0/addons/account/models/account_move.py#L1889

phuctranfxvn commented:
> @phuctranerp Hello, can you please cherry-pick #1492 ?
> I guess the domain needs to be changed for v14.0 by using new field `posted_before`: https://github.com/odoo/odoo/blob/14.0/addons/account/models/account_move.py#L1889

Hi @grindtildeath

Yes sure, done

Thanks

grindtildeath commented on sale_automatic_workflow/data/automatic_workflow_data.xml:
```diff
@@ -0,0 +1,98 @@
+    <record id="automatic_workflow_validate_invoice_filter" model="ir.filters">
+        <field name="name">Automatic Workflow Validate Invoice Filter</field>
+        <field name="model_id">account.move</field>
+        <field name="domain">[('state', '=', 'draft'), ('name', '=', '/')]</field>
```
```suggestion
        <field name="domain">[('state', '=', 'draft'), ('posted_before', '=', False)]</field>
```

grindtildeath reviewed (commented):
:wink:

Commit:
[IMP] sale_automatic_workflow: Change to use the new field posted_before

Co-authored-by: Akim Juillerat <akim.juillerat@camptocamp.com>

rousseldenis commented:
@phuctranerp I relaunched Travis as a fix has been merged for sale_order_lot_generator. Could you check pre-commit ?

phuctranfxvn committed:
[IMP] sale_automatic_workflow: refactoring code

phuctranfxvn commented:
> @phuctranerp I relaunched Travis as a fix has been merged for sale_order_lot_generator. Could you check pre-commit ?

I run the precommit and the travis is green now.

dreispt commented:
/ocabot merge nobump
